### DISCUSSION TITLE
Deploying on Cloudflare Pages

### DISCUSSION THREAD

**Discussion by Strongeyed** [swellstores/community#279] [Help] [answered]
Hi team,

Nearly there to deploy on Cloudflare Pages, however getting the following error: 

15:31:33.784 | > nuxt build
-- | --
15:31:33.784 |  
15:32:16.414 | Finished
15:32:16.415 | Note: No functions dir at /functions found. Skipping.
15:32:16.415 | Validating asset output directory
15:32:16.415 | Error: Output directory "dist" not found.
15:32:17.430 | Failed: build output directory not found

_Actions taken:_

- Added to the ENV public key,  store id and URL.
- Tried NPM RUN BUILD 
- Added Node ENV production node npm build && next export

**Can someone maybe give some input on how they have configured their build on CF pages?**
 
<img width="753" alt="Screen Shot 2023-06-19 at 15 35 23" src="https://github.com/swellstores/community/assets/128098520/54368f0b-fff2-4374-9648-2479be43f1d4">

**Comment by logeshswell** [comment, marked as answer]
@Strongeyed If it refers to the origin theme, then the build command should be,
`yarn run build`
and the definition should be,
`yarn sync && nuxt build && nuxt generate`

**Reason:**
`nuxt generate` command generates `dist` directory and the error occurs since it is missing.

**Reply by Strongeyed** [reply]
Hi @logeshswell perfect, that makes sense so `yarn run build` in the main build and then `yarn sync && nuxt build && nuxt generate` in the ENV? Or all of it in one long string in the build?

Would the command be yarn add sync or just yarn sync?

**Reply by logeshswell** [reply]
> all of it in one long string in the build

> just yarn sync

**Reply by Strongeyed** [reply]
@logeshswell im getting an error that the command Yarn Sync can’t be executed, managed to deploy without yarn sync but not sure if that affects the build negatively.

**Reply by logeshswell** [reply]
@Strongeyed The definition of yarn sync is mentioned below from the [origin-theme](https://github.com/swellstores/origin-theme/blob/master/package.json#L97)
```sh
node -r dotenv/config ./build-utils/sync-storefront.mjs
```

If it is run locally before every deployment, then there shouldn't be any negative impact on the build.

**Reply by logeshswell** [reply]
@Strongeyed What's the error do you get on adding yarn sync?

**Reply by Strongeyed** [reply]
@logeshswell 

13:08:58.488 | yarn run v1.22.4
-- | --
13:08:58.557 | $ nuxt build
13:09:46.612 | Done in 48.13s.
13:09:46.894 | yarn run v1.22.4
13:09:46.978 | error Command "sync" not found.
13:09:46.978 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
13:09:46.996 | Failed: build command exited with code: 1
13:09:47.742 | Failed: error occurred while running build command

**Reply by logeshswell** [reply]
I think you are running a very older version of the origin theme as the yarn version seems to be v1.22.4 instead of v1.22.19.
Can you pull and merge the latest changes of origin theme and try it once again?

Also, we introduced a few env variables in the recent release and make sure to include it in your deployment following the readme instructions.

**Reply by Strongeyed** [reply]
@logeshswell great I think I've managed to do a pull request from the origin theme repo. Only issue is that my logo is now not presenting in the navigation bar anymore, should be an easy fix?

Just wondering if you could initiate the same on our swell hosted storefront (which is currently the live site) as I pulled the outdated repo from there and I don't have a clue how to do this to the remote repo of Swell.

Thanks for all your work @logeshswell 

*Update* I now see that multiple settings changed such as 'quick add to cart' functionality, social icons gone etc. is there an easy fix to resolve this? 

**_Also the build log reports:_**

13:45:35.557 | [warn] Module `@nuxtjs/stylelint-module` is disabled due to incompatibility issues:
-- | --
13:45:35.557 | - [bridge] Nuxt bridge is required
13:46:05.971 | [warn] Module `@nuxtjs/stylelint-module` is disabled due to incompatibility issues:
13:46:05.972 | - [bridge] Nuxt bridge is required

Cheers

Robin

**Reply by logeshswell** [reply]
@Strongeyed I updated your storefront with the latest code changes from the origin theme and also published it.

**Reply by Strongeyed** [reply]
> @Strongeyed I updated your storefront with the latest code changes from the origin theme and also published it.
> 
> 

You are an absolute asset @logeshswell thank you!

**Reply by Strongeyed** [reply]
@logeshswell After the update I ran into a couple of problems, one of them being that my logo, social links and settings are removed at random intervals. The other being that I can't publish anymore because my ENV is missing the SWELL_STOREFRONT_ID.

Error: Missing SWELL_STOREFRONT_ID
at file:///app/build-utils/sync-storefront.mjs:15:11
at Array.forEach ()
at file:///app/build-utils/sync-storefront.mjs:13:17
at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
at async Promise.all (index 0)
at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
at async loadESM (node:internal/process/esm_loader:91:5)
at async handleMainPromise (node:internal/modules/run_main:65:12)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


Build failed!

**Reply by Strongeyed** [reply]
Not sure if these dependency issues have anything to do with the following errors whilst publishing:

warning "@nuxtjs/eslint-config > [eslint-config-standard@16.0.3](mailto:eslint-config-standard@16.0.3)" has incorrect peer dependency "eslint@^7.12.1".
warning "@nuxtjs/eslint-config > [eslint-config-standard@16.0.3](mailto:eslint-config-standard@16.0.3)" has incorrect peer dependency "eslint-plugin-promise@^4.2.1 || ^5.0.0".
warning "@nuxtjs/stylelint-module > [stylelint-webpack-plugin@4.1.1](mailto:stylelint-webpack-plugin@4.1.1)" has incorrect peer dependency "webpack@^5.0.0".
warning "@nuxtjs/stylelint-module > [vite-plugin-stylelint@4.3.0](mailto:vite-plugin-stylelint@4.3.0)" has unmet peer dependency "vite@^2.0.0 || ^3.0.0 || ^4.0.0".
warning " > [stylelint-config-prettier@9.0.5](mailto:stylelint-config-prettier@9.0.5)" has incorrect peer dependency "stylelint@>= 11.x < 15".
warning "stylelint-config-recommended-vue > [stylelint-config-recommended@9.0.0](mailto:stylelint-config-recommended@9.0.0)" has incorrect peer dependency "stylelint@^14.10.0".
[5/5] Building fresh packages...
$ husky install
fatal: not a git repository (or any of the parent directories): .git
Done in 38.81s.

**Reply by Strongeyed** [reply]
@logeshswell In addition to this the page takes minutes to load on the frontend, Google is even giving me a crawl report with a 404 error for the frontend.

**Reply by Strongeyed** [reply]
@logeshswell There's actually quite a couple of more issues now including sitemaps not being generated, Google Tag Manager ID not loading (dropped all traffic exactly 4 days ago), and some settings changing (disappearing) without any reason such as logo not showing, social media links not showing etc. Not sure if the team has notified you so thought I'd also reach out here. **To clarify I'm still hosting through Swell.**

**Reply by logeshswell** [reply]
@Strongeyed As I [mentioned earlier](https://github.com/orgs/swellstores/discussions/279#discussioncomment-6241924), your storefront was running on a significantly older theme version and it wasn't compatible with the latest storefront changes.

So, I updated it to the latest version and as a result, I can see that the default configurations were applied to your store. For example, the logo was absent, **Quick Add to cart** functionality was enabled though it was disabled, etc.

You can update these changes again using the Swell theme editor or via code in the `master` and not in `production` branch.

**Reason:**
Swell stopped the support to manage `master` branch for dev environment and `production` branch for live environment during the notice of storefront hosting deprecation.

Swell uses only `master` branch for both dev and live environment. So, I switched your live store to use `master` branch during the theme version update.

So, it is recommended to switch to `master` branch to make any changes in the live store. You can also use Swell editor to make the live changes and **Publish** button is not required as the changes will be reflected in real-time.

> Error: Missing SWELL_STOREFRONT_ID

This error is applicable only in `production` branch which is not an issue now, as we don't use it anymore.

**Reply by logeshswell** [reply]
> sitemaps not being generated

What was the previous behavior with sitemaps?

> Google Tag Manager not loading

I recovered your previous store configurations and you should see GTM working fine as expected now. Can you check and confirm?

**Reply by Strongeyed** [reply]
> So, it is recommended to switch to master branch to make any changes in the live store. You can also use Swell editor to make the live changes and Publish button is not required as the changes will be reflected in real-time.

The changes I am making are in the swell editor, the issue is not that the changes are not published, the issue is that the changes I make are changing. What I have done is that I have made the amendments once more and I'll monitor it during the day to keep you up to date.

> What was the previous behavior with sitemaps?

Previously sitemaps would be generated for all my products and with the link of my verified domain, there would be a sitemap referring to all the products listed on my website. My sitemap used to have over 825+ entries, and now I have 822 errors in my sitemap submission. This is due to only 4 pages being generated in the sitemap currently.

> So, it is recommended to switch to master branch to make any changes in the live store. You can also use Swell editor to make the live changes and Publish button is not required as the changes will be reflected in real-time.

Good to know we are now only on the master branch, which should not be an issue for me as I use the storefront editor at this stage.

> I recovered your previous store configurations and you should see GTM working fine as expected now. Can you check and confirm?

The GTM is not working as I can't find my tag ID in the source code, Google tag assistant is not confirming it finds any tags either. This used to be fully functional without any issues.

**Reply by Strongeyed** [reply]
@logeshswell feel free to have a look on my environment that I hosted on Cloudflare under Synthesizer.pages.dev that's a lot faster to load, and still generates the sitemaps the old way (wrong links but at least you can see all the links are generated)

**The only problem with this hosted storefront is that it's not 'up-to-date'**, the example of the sitemaps can be found under synthesizer.pages.dev/sitemap.xml

**Reply by logeshswell** [reply]
@Strongeyed Awesome, can you restart your Cloudflare deployment by rebuilding it?
Let's confirm whether GTM is also recovered after the restart.

**Reply by logeshswell** [reply]
If it's working fine, then you can create a new custom storefront in the Swell admin dashboard.
Then, you can ping me so that I can migrate the configs from current storefront to your new storefront.
After that, you can update the env (`SWELL_STOREFRONT_ID`) in your Cloudflare deployment and restart it once again.

**Reply by Strongeyed** [reply]
> @Strongeyed Awesome, can you restart your Cloudflare deployment by rebuilding it?
> 
> Let's confirm whether GTM is also recovered after the restart.

@logeshswell have re-built it, remember this publication is from my GitHub private repository as I cloned it from my Swell repo. Would you need access to it?

I have also created the new storefront in the Swell backend. 

Please be aware that there are a couple of differences between the two repo's (Swell remote & my repo) including different pages etc.

**Reply by logeshswell** [reply]
@Strongeyed I just found a bug in the origin theme, as you mentioned that the storefront is not up-to-date.
The issue is that the default configuration of the storefront is overwritten every time the build is updated.
So, your logo changes, promo bar, social icons and GTM tags are missing for every new deployment.

I found a temporary fix for it by passing `-s defaults` in the sync script as follows:

<img width="1200" alt="Screenshot 2023-06-27 at 2 34 01 AM" src="https://github.com/swellstores/community/assets/100766131/ff102f16-a9e2-4f79-8939-6940b8e87179">

I pushed the change in the swell-remote repo. Can you update the same in your private-repo as well?

**Reply by Strongeyed** [reply]
@logeshswell yes no problem I will do a pull origin/master into the private repo

**Reply by logeshswell** [reply]
@Strongeyed FYI, I migrated the present storefront configs to your new storefront (2eGwGn1Z). You can use it as `SWELL_STOREFRONT_ID` env in Cloudflare deployment.

**Reply by Strongeyed** [reply]
> @Strongeyed FYI, I migrated the present storefront configs to your new storefront (2eGwGn1Z). You can use it as `SWELL_STOREFRONT_ID` env in Cloudflare deployment.

I'm rebuilding now, I'll keep you posted on how we go. Just to clarify, if I would now make changes in my storefront editor in the backend would they be reflected to my frontend still?

**Reply by Strongeyed** [reply]
@logeshswell successfully deployed to Cloudflare now, can you please check at Synthesizer.pages.dev?

**Reply by logeshswell** [reply]
> if I would now make changes in my storefront editor in the backend would they be reflected to my frontend still?

@Strongeyed It will be reflected in your frontend only if you redeploy it since the production build is a static build.

**Reply by Strongeyed** [reply]
@logeshswell Currently on my live site if anyone presses checkout (in the cart) they are forwarded to a '404 not found' now which results in customers not being able to process their orders, could you please assist? It's after I did the migration to my self-hosted storefront.

**Reply by logeshswell** [reply]
@Strongeyed Can you create a CNAME record with your custom domain (synthesizer.nz) pointing to `synthesizer-new-zealand.apps.swellstores.com`
Then, it is good to be verified by available [CNAME lookup](https://dnschecker.org/cname-lookup.php?query=synthesizer.nz&dns=cloudflare) sites.
Once the setup is successful, you should see the checkout working fine as expected.

**Reply by logeshswell** [reply]
@Strongeyed For the meantime, I just made the checkout redirect to `https://synthesizer-new-zealand.swell.store/checkout` which is the default behavior unless the custom domain is configured properly.

**Reply by Strongeyed** [reply]
@logeshswell Just confirming two things, only the www should point to synthesizer-new-zealand.apps.swellstores.com and the @ (root) should point to synthesizer.pages.dev ? 

I'm now getting a SSL Error on the checkout page with either,

- Cloudflare (orange cloud) on: Invalid SSL Certificate > You Working, Sydney Working, Host Error
- Cloudflare (orange cloud) off: Your connection is not private, NET::ERR_CERT_AUTHORITY_INVALID

**Reply by logeshswell** [reply]
@Strongeyed The issue is the domain (synthesizer.pages.dev) is not added to the Swell system, while synthesizer.nz & www.synthesizer.nz are added but DNS configuration is removed later.
So, it is recommended to setup custom domain to your cloudflare deployment, so that synthesizer.nz will render the live site deployed in cloudflare.
Secondly, I need to configure it in our backend to enable the same custom domain for swell-hosted checkout as well, for which you need to add CNAME record and update me.

**Reply by Strongeyed** [reply]
@logeshswell understood I have now changed the CNAME record (www) to reflect the changes and verified them through the DNS Checker. The cloudflare (orange cloud) is currently off, let me know if you want me to turn it on.

**Reply by Strongeyed** [reply]
> So, it is recommended to setup custom domain to your cloudflare deployment, so that synthesizer.nz will render the live site deployed in cloudflare.

I have now also added synthesizer.nz as custom domain to my cloudflare deployment.

<img width="356" alt="Screen Shot 2023-06-27 at 18 13 08" src="https://github.com/swellstores/community/assets/128098520/ef3a66f7-7b33-4d15-bb36-eca6d2a6cddc">

**Reply by logeshswell** [reply]
> The cloudflare (orange cloud) is currently off

Can you turn it on?

**Reply by Strongeyed** [reply]
@logeshswell it's turned on now.

**Reply by logeshswell** [reply]
@Strongeyed Awesome, your Cloudflare storefront is running in https://synthesizer.nz/
Swell-hosted checkout is running on https://www.synthesizer.nz/checkout/...

Sample checkout URL: https://www.synthesizer.nz/checkout/774efb0d229a6053dc391ca9a38c91eb/shipping

**Reply by logeshswell** [reply]
@Strongeyed Is there anything else that you need my assistance?
Make sure to apply [this fix](https://github.com/orgs/swellstores/discussions/279#discussioncomment-6286599) in your private repo and redeploy it.

**Reply by Strongeyed** [reply]
@logeshswell I said it once before but you're awesome! Was just wondering if you can explain to me how I can make changes through the storefront editor, if we solve this issue then at least I can update my logo, social links and Google Tag Manager etc.

What I have already done is changed the links, logo and relevant details in my store editor in the Swell backend. What are my next steps? Just re-deploy my build through Cloudflare pages? I have tried this a couple of times but nothing seems to happen.

**Reply by logeshswell** [reply]
@Strongeyed Can you confirm if [this fix](https://github.com/orgs/swellstores/discussions/279#discussioncomment-6286599) is merged into your private repo through which Cloudflare is deployed?

**Reply by logeshswell** [reply]
If the fix is applied, then please follow the below next steps:
1. Delete the swell-hosted storefront: https://synthesizer-new-zealand.swell.store/admin/storefronts/vGfMtsXq
2. You can edit the theme using the storefront editor: https://synthesizer-new-zealand.swell.store/admin/storefronts/2eGwGn1Z/editor
3. Whenever the theme is edited, you should redeploy your Cloudflare production build.

**Reply by logeshswell** [reply]
> I have tried this a couple of times but nothing seems to happen

Can you share the latest deployment logs?

**Reply by Strongeyed** [reply]
@logeshswell Deploying the fix as we speak, fingers crossed it will work, otherwise I will share the deployment logs with you, no problem!
Will also follow the above steps:

> If the fix is applied, then please follow the below next steps:

>Delete the swell-hosted storefront: https://synthesizer-new-zealand.swell.store/admin/storefronts/vGfMtsXq
>You can edit the theme using the storefront editor: https://synthesizer-new->zealand.swell.store/admin/storefronts/2eGwGn1Z/editor
>Whenever the theme is edited, you should redeploy your Cloudflare production build.

**Reply by Strongeyed** [reply]
@logeshswell We are all good I think, all changes were deployed and I have implemented the fix. Thank you so much for your time!

**Comment by Strongeyed** [comment]
Hi @logeshswell I have an issue deploying once again on Cloudflare. I did not make any changes to the build plan or my own hosted repo, so wondering why this happens?



11:00:14.350 | - [bridge] Nuxt bridge is required
-- | --
11:00:15.823 |  
11:00:15.824 | ERROR  Could not resolve routes
11:00:15.824 |  
11:00:15.824 |  
11:00:15.825 | FATAL  Cannot destructure property 'results' of 'initialFetch' as it is null.
11:00:15.825 |  
11:00:15.825 | at paginateThrough (modules/swell/utils/getRoutes.js:66:11)
11:00:15.825 | at runMicrotasks (<anonymous>)
11:00:15.825 | at processTicksAndRejections (node:internal/process/task_queues:96:5)
11:00:15.826 | at async getRoutes (modules/swell/utils/getRoutes.js:9:3)
11:00:15.826 | at async Generator.initRoutes (node_modules/@nuxt/generator/dist/generator.js:131:26)
11:00:15.826 | at async Generator.generate (node_modules/@nuxt/generator/dist/generator.js:74:20)
11:00:15.826 | at async generate$1 (node_modules/@nuxt/cli/dist/cli-generate.js:46:22)
11:00:15.826 | at async Object.run (node_modules/@nuxt/cli/dist/cli-generate.js:284:7)
11:00:15.826 | at async NuxtCommand.run (node_modules/@nuxt/cli/dist/cli-index.js:413:7)
11:00:15.827 |  
11:00:15.829 |  
11:00:15.830 | ╭──────────────────────────────────────────────────────────────────────────────╮│                                                                              ││   ✖ Nuxt Fatal Error                                                         ││                                                                              ││   TypeError: Cannot destructure property 'results' of 'initialFetch' as it   ││   is                                                                         ││   null.                                                                      ││                                                                              │╰──────────────────────────────────────────────────────────────────────────────╯
11:00:15.830 |  
11:00:15.874 | Failed: build command exited with code: 1
11:00:16.822 | Failed: error occurred while running build command

**Reply by logeshswell** [reply]
@Strongeyed This error may come when you use the incorrect credentials from a different storefront but in the same swell store.
1. Can you check whether you use credentials from [this page](https://synthesizer-new-zealand.swell.store/admin/storefronts/2eGwGn1Z?tab=devtools)?
2. Do you receive the same error on your local machine?

Because, I tried `yarn run build` in my local machine with your swell store and didn't encounter any issues while following the same credentials mentioned on the page.

**Reply by Strongeyed** [reply]
@logeshswell This site has been deployed for months now with the same credentials, I re-deployed it on many occasions and have never changed the credentials in the ENV environment.

Can you let me know if anything changed and which ENV variables need to be there for the origin theme, im still using SWELL_PUBLIC_KEY, SWELL_SECRET_KEY, SWELL_STOREFRONT_ID, SWELL_STORE_ID, SWELL_STORE_URL

**Reply by logeshswell** [reply]
@Strongeyed The mentioned ENV variables look good to me. Can you help me understand which command exactly throw this error?
Is it `nuxt generate`?

**Reply by Strongeyed** [reply]
@logeshswell On one of my other stores I forked the swell origin theme repo directly (it's the latest version) but I get this when I try and deploy it with the same settings on Cloudflare, any ideas?:


08:33:26.220 | ➤ YN0002: │ nuxt@npm:2.15.8 doesn't provide consola (p2d844), requested by @nuxt/components
-- | --
08:33:26.220 | ➤ YN0060: │ swell-origin-theme@workspace:. provides eslint (p44ffd) with version 8.16.0, which doesn't satisfy what @nuxtjs/eslint-config and some of its descendants request
08:33:26.220 | ➤ YN0060: │ swell-origin-theme@workspace:. provides stylelint (pd70aa) with version 15.6.2, which doesn't satisfy what stylelint-config-prettier requests
08:33:26.220 | ➤ YN0060: │ swell-origin-theme@workspace:. provides stylelint (p8d06d) with version 15.6.2, which doesn't satisfy what stylelint-config-recommended-vue and some of its descendants request
08:33:26.220 | ➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
08:33:26.225 | ➤ YN0000: └ Completed in 14s 107ms
08:33:26.347 | ➤ YN0000: ┌ Post-resolution validation
08:33:26.347 | ➤ YN0028: │ The lockfile would have been modified by this install, which is explicitly forbidden.
08:33:26.347 | ➤ YN0000: └ Completed
08:33:26.348 | ➤ YN0000: Failed with errors in 14s 235ms
08:33:26.418 | Error: Exit with error code: 1
08:33:26.418 | at ChildProcess.<anonymous> (/snapshot/dist/run-build.js)
08:33:26.419 | at Object.onceWrapper (node:events:652:26)
08:33:26.419 | at ChildProcess.emit (node:events:537:28)
08:33:26.420 | at ChildProcess._handle.onexit (node:internal/child_process:291:12)
08:33:26.427 | Failed: build command exited with code: 1
08:33:27.336 | Failed: error occurred while running build command

**Reply by logeshswell** [reply]
@Strongeyed I got to know that your Cloudflare deployment is successful now after updating the build directory to `/dist`.
Can you please confirm the same?

**Reply by Strongeyed** [reply]
> @Strongeyed This error may come when you use the incorrect credentials from a different storefront but in the same swell store.
> 
> 1. Can you check whether you use credentials from [this page](https://synthesizer-new-zealand.swell.store/admin/storefronts/2eGwGn1Z?tab=devtools)?
> 
> 2. Do you receive the same error on your local machine?
> 
> 
> 
> Because, I tried `yarn run build` in my local machine with your swell store and didn't encounter any issues while following the same credentials mentioned on the page.

Hi Logesh,

Yes thank you for your support, I just wanted to ask if you could have a look at my other store as well which I’ll deploy with the same settings, the store ID is ‘your-store’

**Reply by logeshswell** [reply]
@Strongeyed Yes sure, if you encounter any issues, just drop a text here (or) to the support team with the screen recordings like previously. I can assist you at the earliest.

But, please don't post it in both places as we get confused with your inputs sometimes.

**Reply by Strongeyed** [reply]
@logeshswell Sure, I completely understand. I will reach out through one channel at a time to avoid confusion. I do think at this moment I can manage the deployment of the other store, if I need you I will sing out!
